packages/@glimmer/runtime/lib/environment.ts

```diff
@@ -537,13 +677,13 @@ export function JitRuntime(
   };
 }
 
-export function JitRuntimeFromProgram(
-  document: SimpleDocument,
+export function JitRuntimeFromProgram<R, E>(
+  options: EnvironmentOptions,
   program: RuntimeProgram,
-  resolver: RuntimeResolverDelegate = {},
-  delegate: RuntimeEnvironmentDelegate = {}
-): JitRuntimeContext {
-  let env = new RuntimeEnvironment(document, new RuntimeEnvironmentDelegateImpl(delegate));
+  resolver: RuntimeResolverDelegate<R> = {},
+  delegate: EnvironmentDelegate<E> = {}
```

Comment:
I think ideally we should try to make it so that the only thing in the system that is aware of the delegate setup that we are using _is_ the environment. I don't particularly think that will make much difference here at the moment, but it does isolate unrelated parts of the system (like this one) from _other future_ changes that we might make to the delegate.

tldr; lets pass the environment here instead of the delegate, but seems fine to do in a follow up...